dependencies.yaml

```diff
@@ -135,6 +138,24 @@ dependencies:
         packages:
           - numpy>=1.23,<2.0a0
           - pynvml>=11.4.1
+  depends_on_cupy:
+    common:
+      - output_types: conda
+        packages:
+          - cupy>=12.0.0
+    # NOTE: This is intentionally not broken into groups by a 'cuda_suffixed' selector like
+    #       other packages with -cu{nn}x suffixes in this file.
+    #       All RAPIDS wheel builds (including in devcontainers) expect cupy to be suffixed.
+    specific:
+      - output_types: [requirements, pyproject]
+        matrices:
+          - matrix: {cuda: "12.*"}
+            packages:
+              - &cupy_cu12 cupy-cuda12x>=12.0.0
+          - matrix: {cuda: "11.*"}
+            packages:
+              - cupy-cuda11x>=12.0.0
+          - {matrix: null, packages: [*cupy_cu12]}
```

Comment:
Are we intentionally tackling https://github.com/rapidsai/build-planning/issues/68 here, or is that coincidental?

Reply:
Yep it's intentional. I mentioned it in the PR description, but not in the title, sorry about that. I've edited the title to make that clearer.